zerver/lib/widget.py

```diff
@@ -21,35 +21,61 @@ def get_widget_data(content: str) -> Tuple[Optional[str], Optional[str]]:
     return None, None
 
 
+def parse_poll_extra_data(content: str) -> Any:
+    # This is used to extract the question from the poll command.
+    # The command '/poll question' will pre-set the question in the poll
+    lines = content.splitlines()
+    question = ""
+    options = []
+    if lines and lines[0]:
+        question = lines.pop(0).strip()
+    for line in lines:
+        # If someone is using the list syntax, we remove it
+        # before adding an option.
+        option = re.sub(r"(\s*[-*]?\s*)", "", line.strip(), count=1)
+        if len(option) > 0:
+            options.append(option)
+    extra_data = {
+        "question": question,
+        "options": options,
+    }
+    return extra_data
+
+
+def parse_todo_extra_data(content: str) -> Any:
+    # This is used to extract the task list title from the todo command.
+    # The command '/todo Title' will pre-set the task list title
+    lines = content.splitlines()
+    task_list_title = ""
+    if lines and lines[0]:
+        task_list_title = lines.pop(0).strip()
+    tasks = []
+    for line in lines:
+        # If someone is using the list syntax, we remove it
+        # before adding a task.
+        task_data = re.sub(r"(\s*[-*]?\s*)", "", line.strip(), count=1)
+        if len(task_data) > 0:
+            # a task and its description (optional) are separated
+            # by the (first) `:` character
+            task_data_array = task_data.split(":", 1)
+            tasks.append(
+                {
+                    "task": task_data_array[0].strip(),
+                    "desc": task_data_array[1].strip() if len(task_data_array) > 1 else "",
+                }
+            )
+    extra_data = {
+        "task_list_title": task_list_title,
+        "tasks": tasks,
+    }
+    return extra_data
```

Comment:
`/todo my title` seems to not work when sent from the web app; what am I missing?